src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs

```diff
@@ -616,5 +616,67 @@ public async Task TestInCastExpressionThatMightBeParenthesizedExpression2(bool h
                 await VerifyItemExistsAsync(markup, "nameof");
             }
         }
+
+        [Theory]
+        [InlineData("class")]
+        [InlineData("struct")]
+        public async Task SuggestRequiredInClassOrStruct(string type)
+        {
+            var markup = $$"""
+                {{type}} C
+                {
+                    $$
+                """;
+
+            await VerifyItemExistsAsync(markup, "required");
+        }
+
+        [Fact]
+        public async Task DontSuggestRequiredInInterface()
+        {
+            var markup = $$"""
+                interface I
+                {
+                    public $$
+                """;
+
+            await VerifyItemIsAbsentAsync(markup, "required");
+        }
+
+        [Fact]
+        public async Task DontSuggestRequiredOnStaticMembers()
```

Comment:
we're unfortunately inconsistent, but preference is DoNot, not Dont in a member name.